drivers/overlay/overlay.go

```diff
@@ -187,12 +213,27 @@ func (d *driver) pushLocalEndpointEvent(action, nid, eid string) {
 
 // DiscoverNew is a notification for a new discovery event, such as a new node joining a cluster
 func (d *driver) DiscoverNew(dType discoverapi.DiscoveryType, data interface{}) error {
-	if dType == discoverapi.NodeDiscovery {
+	switch dType {
+	case discoverapi.NodeDiscovery:
 		nodeData, ok := data.(discoverapi.NodeDiscoveryData)
 		if !ok || nodeData.Address == "" {
 			return fmt.Errorf("invalid discovery data")
 		}
 		d.nodeJoin(nodeData.Address, nodeData.Self)
+	case discoverapi.DatastoreUpdate:
+		var err error
+		if d.store != nil {
+			return types.ForbiddenErrorf("cannot accept datastore update: Overlay driver has a datastore configured already")
+		}
+		dsu, ok := data.(discoverapi.DatastoreUpdateData)
+		if !ok {
+			return types.InternalErrorf("incorrect data in datastore update notification: %v", data)
+		}
+		d.store, err = datastore.NewDataStoreFromUpdate(datastore.GlobalScope, dsu)
```

Comment:
Still in `d.configure()`. I have retained the existing lazy logic where the programming is done on first network create.